src/main/java/jp/co/soramitsu/sora/util/Bencoder.java

```diff
@@ -0,0 +1,224 @@
+package jp.co.soramitsu.sora.util;
+
+import static java.nio.charset.StandardCharsets.US_ASCII;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import javax.validation.constraints.NotNull;
+
+// origin:
+// https://raw.githubusercontent.com/raindev/benjamin/master/src/main/java/org/benjamin/Bencoder.java
+
+/**
+ * NOTE(@warchant): this is NOT original bencode.
+ *
+ * This is bencode, except this:
+ * <ul>
+ * <li> This version can encode Boolean, while original can not. "true" is 'B', "false" is 'b' </li>
+ * <li> If `null` is present in the map as KEY, then it is encoded as STRING "null". </li>
+ * <li> If `null` is present in the map as VALUE, then (key,value) is ignored. </li>
+ * <li> If `null` is present in the list, it is encoded as N. </li>
+ * </ul>
+ *
+ * spec for boolean: "true" is B, "false" is b
+ */
+
+/**
+ * Bencode data encoder.
+ */
+public class Bencoder {
```

Comment:
why do in this class we always return "this" instead of void?

Reply:
to do this, as example:

```
bencoder
  .encode(a)
  .encode(b);